foyer/general_forcefield.py

```diff
@@ -0,0 +1,539 @@
+        top : gmso.Topology, or mb.Compound
+            Molecular Topology to apply the force field to
+        references_file : str, optional, defaut=None
+            Name of file where force field refrences will be written
+            (in Bibtex format).
+        use_residue_map : bool, optional, default=True
+            Options to speed up if there are a lot of repeated
+            subtopology within the topology (assuming they all
+            have the same name).
+        assert_bond_params : bool, optional, default=True
+            If True, Foyer will exit if parameters are not found for all
+            system bonds.
+        assert_angle_params : bool, optional, default=True
+            If True, Foyer will exit if parameters are not found for all
+            system angles.
+        assert_dihedral_params : bool, optional, default=True
+            If True, Foyer will exit if parameters are not found for all
+            system dihedrals.
+        assert_improper_params : bool, optional, default=True
+            If True, Foyer will exit if parametes are not found for all
+            system impropers.
+        combining_rule : str, optional, default='geometric'
+            The combining rule of the system, stored as an attribute of the
+            GMSO topology.
+        verbose : bool, optional, default=False
+            If True, Foyer will print debug-level information about notable or
+            potential problematic details it encounters.
+        backend : str, optional, default='gmso'
+            Name of the backend used to store all the Types' information.
+            At this point, 'gmso' is the only valid backend, but this set up
+            allow future backend to be implemented easier.
+        """
+        if self.atomTypeDefinitions == {}:
+            raise FoyerError('Attempting to atom-type using a forcefield '
+                'with no atom type definitions.')
+
+        if backend == 'gmso':
+            if not isinstance(top, gmso.Topology):
+                mb = import_('mbuild')
+                tmp_top = mb.load(top)
+                top = gmso.external.from_mbuild(top)
+
+            assert isinstance(top, gmso.Topology)
+            typemap = self._run_atomtyping(top,
+                    use_residue_map=use_residue_map,
+                    **kwargs)
+
+        return self._parametrize(top=top, typemap=typemap,
+            references_file=references_file,
+            assert_bond_params=assert_bond_params,
+            assert_angle_params=assert_angle_params,
+            assert_dihedral_params=assert_dihedral_params,
+            assert_improper_params=assert_improper_params,
+            verbose=verbose,
+            backend=backend,
+            combining_rule=combining_rule,
+            *args, **kwargs)
+
+    def _run_atomtyping(self, top, use_residue_map=True, **kwargs):
+        """Atomtype the topology
+
+        Parameters
+        ----------
+        top : gmso.Topology
```

Comment:
"Molecular Topology to find atom types for. " sounds a little weird to me.  Maybe "Molecular Topology which is utilized to determine atom types?

In "Spped up options for duplicates subtopology" , Speed is spelled wrong.  
-Also a suggestion (not required), it may be more informative for new users to list "A speed-up option that utilizes previously atom typed molecules as a template for future atom typing of identical molecules, instead of reevaluating them each individually."